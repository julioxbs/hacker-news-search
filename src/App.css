* {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
}

.container {
    max-width: 1000px;
    margin: 0 auto;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

input {
    padding: 8px 12px;
}

form {
    margin-top: 1.5rem;
}

button {
    padding: 8px 12px;
    margin-left: 10px;
}

.buttons {
    margin-top: 1rem;
}

ul {
    margin: 1.5rem 0;
}