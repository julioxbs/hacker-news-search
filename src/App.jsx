import { Component } from "react";
import Search from './Components/Search';
import Loading from "./Components/Loading";
import './App.css';

export default class MyApp extends Component {
    _isMounted = false;

    constructor(props) {
        super(props);

        this.state = {
            result: null,
            searchTerm: 'redux',
            page: 0,
            isLoading: false,
        }

        this.handleInput = this.handleInput.bind(this);
        this.setSearch = this.setSearch.bind(this);
        this.nextPage = this.nextPage.bind(this);
        this.prevPage = this.prevPage.bind(this);
    }

    componentDidMount() {
        this._isMounted = true;
        const { searchTerm, page } = this.state
        this.getDataApi(searchTerm, page);
    }

    componentWillUnmount() {
        this._isMounted = false;
    }

    async getDataApi(searchTerm, page) {
        try {
            let response = await fetch(`http://hn.algolia.com/api/v1/search?query=${searchTerm}&tags=story&page=${page}`);
            let data = await response.json();
            this._isMounted && this.setState({ result: data, isLoading: (data ? true : false) });
        } catch (error) {
            console.log(error);
        }
    }

    setSearch(e) {
        this.setState({ searchTerm: e.target.value });
    }

    handleInput(e) {
        const { searchTerm, page } = this.state;
        e.preventDefault();
        this.getDataApi(searchTerm, page);
    }

    nextPage() {
        const { searchTerm, page } = this.state;
        this.setState({ page: page + 1 });
        this.getDataApi(searchTerm, page);
    }

    prevPage() {
        const { searchTerm, page } = this.state;
        this.setState({ page: page - (page > 0 ? 1 : 0) });
        this.getDataApi(searchTerm, page);
    }

    render() {
        const { result, isLoading } = this.state;
        return (
            <div className="container">
                <Search
                    onSubmit={this.handleInput}
                    onChange={this.setSearch}>
                    Pesquisar...
                </Search>

                <ul>
                    {result ? result.hits.map((items, i) =>
                        <li key={i}>{items.title}</li>
                    ) : null}
                </ul>

                <div className="buttons">
                    {isLoading
                        ? <>
                            <button onClick={this.prevPage}>
                                Voltar
                            </button>

                            <button onClick={this.nextPage}>
                                Proximo
                            </button>
                        </> : <Loading />

                    }
                </div>
            </div>
        );
    }
}

