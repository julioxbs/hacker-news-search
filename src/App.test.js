import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer'
import App from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <App/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })  
})

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

