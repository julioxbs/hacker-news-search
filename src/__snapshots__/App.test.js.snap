// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Pesquisar..."
      type="text"
    />
    <button
      type="submit"
    >
      Pesquisar
    </button>
  </form>
  <ul />
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
    >
      Voltar
    </button>
    <button
      onClick={[Function]}
    >
      Proximo
    </button>
  </div>
</div>
`;
